test/test_dotu_device.cc

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) 2017-2022, University of Tennessee. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the BSD 3-Clause license. See the accompanying LICENSE file.
+
+#include "test.hh"
+#include "cblas_wrappers.hh"
+#include "lapack_wrappers.hh"
+#include "blas/flops.hh"
+#include "print_matrix.hh"
+
+// -----------------------------------------------------------------------------
+template< typename Tx, typename Ty >
```

Comment:
Ditto comments from test_dot_device.cc. I would also `meld test_dot_device.cc test_dotu_device.cc`. The changes should be minimal.